Feature/#276 proxy subdomain

AlviseSembenico opened the pull request:
Here we go, I added the feature to strip the url in order to fetch some resource correctly but I am not sure if it is how it is meant to be, we can discuss it on Monday.
Thanks

Commit:
Model change and chatch subdomain

Commit:
Fix dependency

Commit:
Move to subdomain

Commit:
Fix reverse in serializer

Commit:
:white_check_mark: fix test with subdomain

Commit:
Correct order urls, fix None in id

Commit:
Strip of the relative url

Commit:
Merge branch 'develop' into feature/#276-proxy-subdomain

alextreme commented on requirements/base.txt:
```diff
@@ -5,6 +5,7 @@
 #    pip-compile --output-file requirements/base.txt requirements/base.in
 #
 -e git+https://github.com/maykinmedia/gemma-zds-client-3.4.git@e826a2fa111100d33d7c164a7bdf95d07c3b916c#egg=gemma-zds-client
+-e git+https://github.com/maykinmedia/django-subdomains.git#egg=django-subdomains
```
@AlviseSembenico ensure you pin this against a commit, or make the use of master explicit if you're brave

alextreme commented:
@AlviseSembenico could you get this working with subdomains of the vorm:

proxy1234-vng-staging.maykin.nl ?

*.maykin.nl is already pointing to rachel, and if you see in deployments/nginx-*.conf we have a SSL wildcard certificate installed. So we'd only have to handle all servers with a specific postfix (if Nginx allows this, I'm not sure)

We can also get a certificate for proxy1234.vng-staging.maykin.nl if necessary but from experience DNS wildcards at multiple levels will get tricky.

Commit:
Handle partial subdomain

AlviseSembenico commented:
I tryied to deploy it in the test but clearly we have to unlock it othewise the authentication is asked by nginx

Commit:
Merge branch 'develop' into feature/#276-proxy-subdomain

Commit:
Nginx test config

Commit:
Merge branch 'feature/#276-proxy-subdomain' of github.com:maykinmedia/api-testvoorziening into feature/#276-proxy-subdomain

Commit:
Merge migrations

Commit:
Fix server name

Commit:
Fix server name

Fix server name

Remove pointer name

add quotes

Add allowed host

Change wildcard

:bug: fix missing session

Enforce https for reverse

Commit:
Fix server name

Commit:
Remove pointer name

Commit:
add quotes

Commit:
Add allowed host

Commit:
Change wildcard

Commit:
:bug: fix missing session

Commit:
Enforce https for reverse

Commit:
Merge branch 'feature/#276-proxy-subdomain' of github.com:maykinmedia/api-testvoorziening into feature/#276-proxy-subdomain

Commit:
Fix url substitution with subdomain

Commit:
Url match https

Commit:
Remove useless url reference

Commit:
Fix tests

Commit:
:recycle: set right appname

Commit:
Merge branch 'develop' into feature/#276-proxy-subdomain

Commit:
Update test in develop

Commit:
Fix http host test

AlviseSembenico commented:
@alextreme Everything operative

alextreme commented:
@AlviseSembenico great! Feel free to roll this out on vng-test
